### PR TITLE
BDS-46: Fail if BBMD registration fails

### PR DESCRIPTION
**Pull request by gofling**

### CODE REVIEW

File: src/bvlc.c [hunk 1]
```diff
@@ -1194,7 +1199,7 @@ uint16_t bvlc_receive(
     uint16_t i = 0;
     bool status = false;
     uint16_t time_to_live = 0;
-
+
```
Comment (iain-anderson) [inline src/bvlc.c]: remove this please

Comment (gofling) [reply src/bvlc.c]: Done.

File: ports/linux/dlmstp.c [hunk 2]
```diff
@@ -144,7 +144,6 @@ void dlmstp_cleanup(
     pthread_mutex_destroy(&Received_Frame_Mutex);
     pthread_mutex_destroy(&Receive_Packet_Mutex);
     pthread_mutex_destroy(&Master_Done_Mutex);
-    pthread_cancel(hThread);
```
Comment (iain-anderson) [inline ports/linux/dlmstp.c]: this still gets created in dlsmtp_init ?

File: src/bvlc.c [hunk 3]
```diff
@@ -1255,6 +1261,10 @@ uint16_t bvlc_receive(
             (void) decode_unsigned16(&npdu[4], &result_code);
             BVLC_Result_Code = (BACNET_BVLC_RESULT) result_code;
             debug_printf("BVLC: Result Code=%d\n", BVLC_Result_Code);
+            /* Set the BBMD registration bool to successfull, signal the condition variable and unlock the mutex */
+            bbmd_reg_success = true;
+            pthread_cond_signal (&cond);
+            pthread_mutex_unlock (&mutex);
```
Comment (iain-anderson) [inline src/bvlc.c:1268]: This should have been locked somewhere

File: ports/linux/dlmstp.c (hunk 2 above)
Comment (gofling) [reply ports/linux/dlmstp.c]: Re-added this

File: src/bvlc.c (hunk 3 above)
Comment (gofling) [reply src/bvlc.c:1268]: Locked before setting bbmd_reg_success to true

**Review by iain-anderson — APPROVED** [review]
LGTM

**Pull request merged**